vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/jsonItem/ArrayPointable.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.vxquery.datamodel.accessors.jsonItem;
+
+import org.apache.hyracks.api.dataflow.value.ITypeTraits;
+import org.apache.hyracks.data.std.api.IPointable;
+import org.apache.hyracks.data.std.api.IPointableFactory;
+import org.apache.hyracks.data.std.primitive.IntegerPointable;
+import org.apache.hyracks.data.std.primitive.VoidPointable;
+import org.apache.vxquery.datamodel.accessors.SequencePointable;
+
+public class ArrayPointable extends SequencePointable {
```

Comment:
It looks like you extend the SequencePointable, but rewrite all the functions? What is not being overridden? Maybe it should just be separate.